tests/linearizability/traffic.go

```diff
@@ -50,34 +50,42 @@ func (t readWriteSingleKey) Run(ctx context.Context, c *recordingClient, limiter
 		default:
 		}
 		// Execute one read per one write to avoid operation history include too many failed writes when etcd is down.
-		err := t.Read(ctx, c, limiter)
+		resp, err := t.Read(ctx, c, limiter)
 		if err != nil {
 			continue
 		}
 		// Provide each write with unique id to make it easier to validate operation history.
-		t.Write(ctx, c, limiter, ids.RequestId())
+		t.Write(ctx, c, limiter, ids.RequestId(), resp)
 	}
 }
 
-func (t readWriteSingleKey) Read(ctx context.Context, c *recordingClient, limiter *rate.Limiter) error {
-	getCtx, cancel := context.WithTimeout(ctx, 20*time.Millisecond)
-	err := c.Get(getCtx, t.key)
+func (t readWriteSingleKey) Read(ctx context.Context, c *recordingClient, limiter *rate.Limiter) (string, error) {
+	getCtx, cancel := context.WithTimeout(ctx, 100*time.Millisecond)
+	resp, err := c.Get(getCtx, t.key)
 	cancel()
 	if err == nil {
 		limiter.Wait(ctx)
 	}
-	return err
+	return resp, err
 }
 
-func (t readWriteSingleKey) Write(ctx context.Context, c *recordingClient, limiter *rate.Limiter, id int) error {
-	putCtx, cancel := context.WithTimeout(ctx, 20*time.Millisecond)
+func (t readWriteSingleKey) Write(ctx context.Context, c *recordingClient, limiter *rate.Limiter, id int, readResponse string) error {
+	putCtx, cancel := context.WithTimeout(ctx, 100*time.Millisecond)
 
 	var err error
 	switch t.pickWriteOperation() {
 	case Put:
 		err = c.Put(putCtx, t.key, fmt.Sprintf("%d", id))
 	case Delete:
 		err = c.Delete(putCtx, t.key)
+	case Txn:
+		var value string
+		if readResponse == "" {
+			value = fmt.Sprintf("%d", id)
+		} else {
+			value = fmt.Sprintf("%s,%d", readResponse, id)
```

Comment:
Have you checked the visualized diagram? I guess it may be hard to read, because the value may be long long string.

Reply:
It would be better to replace the suffix ID instead of always appending an ID.

Reply:
It's manageable as values are overriden/reset by Put and Delete requests.